slick.dataview.js

```diff
@@ -1265,67 +1267,168 @@
       _grid = grid;
       var inHandler;
       selectedRowIds = self.mapRowsToIds(grid.getSelectedRows());
-      var onSelectedRowIdsChanged = new Slick.Event();
 
+      /** @param {Array} rowIds */
       function setSelectedRowIds(rowIds) {
-        if (selectedRowIds.join(",") == rowIds.join(",")) {
-          return;
+        if (rowIds === false) {
+          selectedRowIds = [];
+        } else {
+          if (selectedRowIds.sort().join(",") !== rowIds.sort().join(",")) {
+            selectedRowIds = rowIds;
+          }
         }
-
-        selectedRowIds = rowIds;
-
-        onSelectedRowIdsChanged.notify({
-          "grid": grid,
-          "ids": selectedRowIds,
-          "dataView": self
-        }, new Slick.EventData(), self);
       }
 
       function update() {
-        if (selectedRowIds.length > 0) {
+        if (selectedRowIds.length > 0 && !inHandler) {
           inHandler = true;
           var selectedRows = self.mapIdsToRows(selectedRowIds);
           if (!preserveHidden) {
-            setSelectedRowIds(self.mapRowsToIds(selectedRows));
+            var selectedRowsChangedArgs = {
+              "grid": _grid,
+              "ids": self.mapRowsToIds(selectedRows),
+              "dataView": self
+            };
+            selectedRowChangedFn(selectedRowsChangedArgs);
+            onSelectedRowIdsChanged.notify(Object.assign(selectedRowsChangedArgs, {
+              "filteredIds": self.getAllSelectedFilteredIds(),
+            }), new Slick.EventData(), self);
           }
           grid.setSelectedRows(selectedRows);
           inHandler = false;
         }
       }
 
       grid.onSelectedRowsChanged.subscribe(function (e, args) {
-        if (inHandler) { return; }
-        var newSelectedRowIds = self.mapRowsToIds(grid.getSelectedRows());
-        if (!preserveHiddenOnSelectionChange || !grid.getOptions().multiSelect) {
-          setSelectedRowIds(newSelectedRowIds);
-        } else {
-          // keep the ones that are hidden
-          var existing = $.grep(selectedRowIds, function (id) { return self.getRowById(id) === undefined; });
-          // add the newly selected ones
-          setSelectedRowIds(existing.concat(newSelectedRowIds));
+        if (!inHandler) {
+          var newSelectedRowIds = self.mapRowsToIds(grid.getSelectedRows());
+          var selectedRowsChangedArgs = {
+            "grid": _grid,
+            "ids": newSelectedRowIds,
+            "added": true,
+            "dataView": self
+          };
+          selectedRowChangedFn(selectedRowsChangedArgs);
+          onSelectedRowIdsChanged.notify(Object.assign(selectedRowsChangedArgs, {
+            "filteredIds": self.getAllSelectedFilteredIds(),
+          }), new Slick.EventData(), self);
         }
       });
 
+      selectedRowChangedFn = function (args) {
```

Comment:
in previous PR #689, this used to be `grid.onSelectedRowsChanged.subscribe`, but that was causing a racing condition since the Example Demo also uses `onSelectedRowsChanged`, in summary we need to execute this code here **before** the `onSelectedRowsChanged` in the Example, we can accomplish this by turning this into a function that is called before the `onSelectedRowsChanged.notify` (see other lines of this PR)